File: Java Labs/Patterns On Java/src/com/pattern/mediator/InventorySystem.java
package com.pattern.mediator;

public class InventorySystem implements Component {
    private Mediator mediator;

    @Override
    public void setMediator(Mediator mediator) {
        this.mediator = mediator;
    }

    @Override
    public void notify(String event) {
        System.out.println("Inventory System received notification: " + event);
        // Additional logic related to inventory management...
    }

    public void checkStock(String item) {
        // Inventory-specific logic...
        mediator.notify(this, "Stock checked for item: " + item);
    }
}

File: Python Labs/com/pattern/Mediator/InventorySystem.py
from com.pattern.Mediator.Component import Component


class InventorySystem(Component):
    def __init__(self):
        self.mediator = None

    def set_mediator(self, mediator):
        self.mediator = mediator

    def notify(self, event):
        print(f"Inventory System received notification: {event}")
        # Additional logic related to inventory management...

    def check_stock(self, item):
        # Inventory-specific logic...
        self.mediator.notify(self, f"Stock checked for item: {item}")